docs/DESIGN_DESICIONS/applications_required_resource_access.md

```diff
@@ -0,0 +1,88 @@
+# Design Decision: Auto-Approving Application Roles in `required_resource_access`
+
+**Date:** March 2, 2025  
+**Author:** Sander Blomvågnes  
+
+## Context  
+
+Previously, when using `var.applications.app_name.required_resource_access`, only the `resource_app_id` needed to be provided. However, roles were not automatically assigned, requiring a Global Admin to manually grant the necessary permissions in Entra ID after deployment.  
+
+To enable automatic assignment of permissions, we now require the following details:  
+
+- The **role ID** (e.g., `User.Read.All`)  
+- The **client ID** of the service principal where permissions should be assigned (e.g., Microsoft Graph)  
+- The **object ID** of the service principal where permissions should be assigned. This ID is generated when creating the tenant and differs across tenants, unlike the client ID and role ID.  
+
+## Reasoning  
```

Comment:
Can resoning section be written more concise?
> - Automating permission granting eliminates redundant approval steps, as the same person often approves the pull request and grants permissions in Entra ID.
> - Reducing module resources lowers Terraform Cloud (TFC) costs and improves plan execution time.
> - Users must now provide both client_id and object_id instead of mimicking required_resource_access from the azuread provider. This removes the need for data lookups, cutting TFC costs and redundant API calls.